conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueClientGenerator.java

```diff
@@ -0,0 +1,586 @@
+import com.palantir.dialogue.DialogueOkHttpErrorDecoder;
+import com.palantir.dialogue.EmptyBody;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Exceptions;
+import com.palantir.dialogue.HttpMethod;
+import com.palantir.dialogue.OkHttpErrorDecoder;
+import com.palantir.dialogue.PathTemplate;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Serializer;
+import com.palantir.dialogue.Serializers;
+import com.palantir.tokens.auth.AuthHeader;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Modifier;
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.uri.internal.UriTemplateParser;
+
+// TODO(rfink): Add unit tests for misc edge cases, e.g.: docs/no-docs, auth/no-auth, binary return type.
+
+/** Experimental support for Conjure clients using the https://github.com/palantir/dialogue library. */
+public final class DialogueClientGenerator implements ServiceGenerator {
+
+    private static final Class<?> EMPTY_BODY_CLASS = EmptyBody.class;
+    private static final TypeName EMPTY_BODY = TypeName.get(EMPTY_BODY_CLASS);
+
+    @Override
+    public Set<JavaFile> generate(ConjureDefinition conjureDefinition) {
+        TypeMapper parameterTypes = new TypeMapper(conjureDefinition.getTypes(),
+                new ClassVisitor(conjureDefinition.getTypes(), ClassVisitor.Mode.PARAMETER));
+        TypeMapper returnTypes = new TypeMapper(conjureDefinition.getTypes(),
+                new ClassVisitor(conjureDefinition.getTypes(), ClassVisitor.Mode.RETURN_VALUE));
+        TypeAwareGenerator generator = new TypeAwareGenerator(parameterTypes, returnTypes);
+        return conjureDefinition.getServices().stream()
+                .flatMap(serviceDef -> generator.service(serviceDef).stream())
+                .collect(Collectors.toSet());
+    }
+
+    // TODO(rfink): Split into separate classes: endpoint, interface, impl.
+    private static final class TypeAwareGenerator {
+        private static final String AUTH_HEADER_NAME = "Authorization";
+        private static final String AUTH_HEADER_PARAM_NAME = "authHeader";
+        private static final String MAPPER_FIELD_NAME = "mapper";
+        private static final String REQUEST_VAR_NAME = "_request";
+        private static final String REQUEST_VAR_NAME_BUILDER = "_request_builder";
```

Comment:
curious to know what is the convention in such case. Does the prefix `_` forces us to avoid caml case or is it also acceptable to use "_requestBuilder"

Reply:
should be _requestBuilder, thanks